keras_cv/models/resnet_v2.py

```diff
@@ -99,210 +101,312 @@
 """
 
 
-def BasicBlock(
-    filters, kernel_size=3, stride=1, dilation=1, conv_shortcut=False, name=None
-):
+@keras.utils.register_keras_serializable(package="keras_cv.models.resnet_v2")
+class BasicBlock(keras.layers.Layer):
     """A basic residual block (v2).
+
     Args:
         filters: integer, filters of the basic layer.
-        kernel_size: default 3, kernel size of the bottleneck layer.
-        stride: default 1, stride of the first layer.
-        conv_shortcut: default False, use convolution shortcut if True,
+        kernel_size: integer, kernel size of the bottleneck layer. Defaults to 3.
+        stride: integer, stride of the first layer. Defaults to 1.
+        dilation: integer, the dilation rate to use for dilated convolution. Defaults to 1.
+        conv_shortcut: bool, uses convolution shortcut if `True`. Defaults to `False`.
           otherwise identity shortcut.
```

Comment:
"otherwise identity shortcut" isn't quite true -- really it defaults to identity or pooling, based on stride.
I don't think including all the detail is necessary in this docstring, but maybe we could say "otherwise identity or pooling shortcut, based on stride"

Reply:
(same for BasicBlock)